ion_functions/data/prs_functions.py

```diff
@@ -821,16 +821,17 @@ def anchor_bin_raw_data_to_15s(time, data):
     data[np.isnan(data)] = -999.0  # use any negative value
     mask = (data > 0.0)
     time = time[mask]
-    data = data[mask]
 
     # anchor time-centered bins by determining the start time to be half a bin
     # before the first 'anchor timestamp', which will be an integral number of
     # bin_durations after midnight.
     start_time = np.floor((time[0] - half_bin)/bin_duration) * bin_duration + half_bin
     # calculate elapsed time from start in units of bin_duration.
-    time_elapsed = (time - start_time)/bin_duration
+    elapsed_time = time
+    elapsed_time -= start_time
+    elapsed_time /= bin_duration
```

Comment:
I see the variable name change but curious why you altered the 1-line command to 3 lines. It doesn't look like you changed the result.

Reply:
It's a memory footprint issue. You're absolutely right that the result is the same. But the multi-line version manipulates the structure in place.